### PR TITLE
Detect vercel again after rebranding in gatsby-telemetry

### PR DESCRIPTION
**Pull request by jamo**

### CODE REVIEW

File: packages/gatsby-core-utils/src/ci.ts [hunk 1]
```diff
@@ -4,6 +4,8 @@ const CI_DEFINITIONS = [
   envFromCIandCIName,
   getEnvDetect({ key: `NOW_BUILDER_ANNOTATE`, name: `ZEIT Now` }),
   getEnvDetect({ key: `NOW_REGION`, name: `ZEIT Now v1` }),
+  getEnvDetect({ key: `VERCEL_URL`, name: `Vercel Now` }),
+  getEnvDetect({ key: `NOW_BUILDER`, name: `Vercel Now` }),
```
Comment (muescha) [inline packages/gatsby-core-utils/src/ci.ts:8]: when i look at this doc: the `NOW_BUILDER` is also renamed to `VERCEL_BUILDER`:

https://www.notion.so/Change-Timeline-fab7622780694753a92c0a396cea7eb6

Comment (muescha) [reply packages/gatsby-core-utils/src/ci.ts:8]: cc #24019

Comment (jamo) [reply packages/gatsby-core-utils/src/ci.ts:8]: they still expose NOW_BUILDER 🙈

Comment (jamo) [reply packages/gatsby-core-utils/src/ci.ts:8]: tho why not add both should they one day fix it

Comment (jamo) [reply packages/gatsby-core-utils/src/ci.ts:8]: https://github.com/gatsbyjs/gatsby/pull/25926

Comment (muescha) [reply packages/gatsby-core-utils/src/ci.ts:8]: > Existing projects will get both NOW_ and VERCEL_ env. New projects will only get VERCEL_ env.

there is no info about an expire

Comment (muescha) [reply packages/gatsby-core-utils/src/ci.ts:8]: > they still expose NOW_BUILDER 🙈

only on existing apps

Comment (jamo) [reply packages/gatsby-core-utils/src/ci.ts:8]: I'd create a new app fwiw

Comment (muescha) [reply packages/gatsby-core-utils/src/ci.ts:8]: 👍

**Pull request merged**